calico_mesos/calico_mesos.py

```diff
@@ -601,6 +602,18 @@ class IsolatorException(Exception):
         _log.error(e)
         sys.stdout.write(_error_message(str(e)))
         sys.exit(1)
+    except etcd.EtcdConnectionFailed as e:
+        _log.error(e)
+        if 'ETCD_AUTHORITY' in os.environ:
+            error_message = ("Failed to communicate with etcd at '%s'. "
+                    "Ensure that it is up and running or change ETCD_AUTHORITY"
+                     " environment variable used by the meoss-slave process.") % \
+                             os.environ['ETCD_AUTHORITY']
```

Comment:
You're accessing the environ dictionary twice for the same thing.  Nicer to get it once and reuse.
```
etcd = os.environ.get("ETCD_AUTHORITY")
if etcd:
    print etcd
```

Do we really hit this exception?  I thought the data store wrapped all failures in a DataStoreException.  If not that sounds like a bug in libcalico.